snips-nlu-lib/src/intent_classifier/featurizer.rs

```diff
@@ -84,27 +91,52 @@ impl Featurizer {
     }
 
     fn preprocess_query(&self, query: &str) -> Vec<String> {
-        let tokens = tokenize_light(
-            query,
-            NluUtilsLanguage::from_language(self.language_config.language),
-        );
-        let mut processed_tokens: Vec<String> = if let Some(ref stemmer) = self.stemmer {
-            tokens.iter().map(|t| stemmer.stem(&normalize(t))).collect()
-        } else {
-            tokens.iter().map(|t| normalize(t)).collect()
-        };
-        if let Some(ref clusterer) = self.word_clusterer {
-            processed_tokens.append(&mut get_word_cluster_features(&tokens, clusterer))
-        }
-        processed_tokens.append(&mut get_dataset_entities_features(
-            &tokens,
-            self.stemmer.as_ref(),
+        let language = NluUtilsLanguage::from_language(self.language_config.language);
+        let tokens = tokenize_light(query, language);
+        let word_cluster_features = self.word_clusterer
+            .as_ref()
+            .map(|clusterer| get_word_cluster_features(&tokens, clusterer))
+            .unwrap_or(vec![]);
```

Comment:
Use `unwrap_or_else` instead to avoid to evaluate (and thus instanciate for nothing) the param before we actually need it.